Apsitvarkom.UnitTests/DataAccess/PollutedLocationDTOFileRepositoryTests.cs

```diff
@@ -103,43 +104,107 @@ public async Task GetAllPollutedLocationsDTO_JsonIncludesMultipleInstances_AllIn
+    public void GetAllAsync_ReadFromFile_CouldNotFindSourceFile_Throws()
     {
         var notExistingSourcePath = Guid.NewGuid() + ".json";
-        Assert.Throws<FileNotFoundException>(() => PollutedLocationsDTOFileRepository.FromFile(notExistingSourcePath));
+        Assert.Throws<FileNotFoundException>(() => PollutedLocationDTOFileRepository.FromFile(notExistingSourcePath));
     }
+    #endregion
+
+    #region GetByIdAsync tests
+
+    [Test]
+    [TestCase("b38b9bf6-74f6-4325-8ddf-9defe9bc2994", "3fd9bd2a-90ac-4ae1-baee-3c31b91e91f6", "6ad05412-a6c5-436d-9795-8581b27bfadb", "71009336-9133-47c8-b577-d755c8c371ee")]
+    public async Task GetByIdAsync_JsonIncludesTheRequiredInstance_ReturnsInstance(string id1, string id2, string id3, string id4)
+    {
+        var jsonString =
+            "[" +
+            $"{{\"id\":\"{id1}\"}}," +
+            $"{{\"id\":\"{id2}\"}}," +
+            $"{{\"id\":\"{id3}\"}}," +
+            $"{{\"id\":\"{id4}\"}}" +
+            "]";
+        using var dataManager = PollutedLocationDTOFileRepository.FromContent(jsonString);
+
+        var requestId = id2;
+
+        var instance = await dataManager.GetByIdAsync(requestId);
+
+        Assert.That(instance, Is.Not.Null);
+        Assert.That(instance!.Id, Is.EqualTo(requestId));
+    }
+
+    [Test]
+    [TestCase("b38b9bf6-74f6-4325-8ddf-9defe9bc2994", "3fd9bd2a-90ac-4ae1-baee-3c31b91e91f6", "6ad05412-a6c5-436d-9795-8581b27bfadb", "71009336-9133-47c8-b577-d755c8c371ee")]
+    public async Task GetByIdAsync_NoInstanceWithRequestedIdFound_ReturnsNull(string id1, string id2, string id3, string id4)
+    {
+        var jsonString =
+            "[" +
+            $"{{\"id\":\"{id1}\"}}," +
+            $"{{\"id\":\"{id2}\"}}," +
+            $"{{\"id\":\"{id3}\"}}," +
+            $"{{\"id\":\"{id4}\"}}" +
+            "]";
+        using var dataManager = PollutedLocationDTOFileRepository.FromContent(jsonString);
+
+        var requestId = Guid.NewGuid().ToString();
+
+        var instance = await dataManager.GetByIdAsync(requestId);
+
+        Assert.That(instance, Is.Null);
+    }
+
+    [Test]
+    [TestCase("b38b9bf6-74f6-4325-8ddf-9defe9bc2994", "3fd9bd2a-90ac-4ae1-baee-3c31b91e91f6", "6ad05412-a6c5-436d-9795-8581b27bfadb")]
+    public void GetByIdAsync_SeveralInstanceWithRequestedIdFound_Throws(string id1, string id2, string id3)
+    {
+        var jsonString =
+            "[" +
+            $"{{\"id\":\"{id1}\"}}," +
+            $"{{\"id\":\"{id2}\"}}," +
+            $"{{\"id\":\"{id3}\"}}," +
+            $"{{\"id\":\"{id2}\"}}" +
+            "]";
+        using var dataManager = PollutedLocationDTOFileRepository.FromContent(jsonString);
+
+        var requestId = id2;
+
+        Assert.That(Assert.ThrowsAsync<InvalidOperationException>(async () => await dataManager.GetByIdAsync(requestId))!.Message, 
+            Is.EqualTo("Sequence contains more than one matching element"));
```

Comment:
Why do you need to check with a hard coded string? `Assert.ThrowsAsync` alone will do just what you need – assert if it throws. I don't see a reason to check the exception message